src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        int numberFigure = 4;
+        Figure[] figures = new Figure[numberFigure];
+        for (int a = 0; a < figures.length; a++) {
+            if (a <= figures.length / 2) {
+                figures[a] = new FigureSupplier().getRandomFigure();
```

Comment:
create only one instance of FigureSupplier in thic class

Reply:
not fixed